andela/bp-esa-backend
# #161914907 Fetch new placements from Allocations

#### What does this PR do?
Allow new placements for the past 24 hours to be fetched

#### Description of Task to be completed?
- refactor fetchPlacementsByStatus function to eliminate callbacks and return axios promise
- refactor fetchNewPlacements function to eliminate callbacks and 
- return promise of response data filtered by created_at date

#### How should this be manually tested?
1. In module: `server/modules/allocations.js`,  call `fetchPlacementsByStatus` function while providing the `status` of placement to be fetched, and console.log the promise resolve and reject responses
3. On the terminal, run module `server/modules/allocations.js` using the command: `npx babel-node allocations.js`
2. In module: `server/modules/allocations.js`,  call `fetchNewPlacements` function while providing the `status` and `numberOfdays`, and console.log the promise resolve and reject responses
3. On the terminal, run module `server/modules/allocations.js` using command: `npx babel-node allocations.js`
3. Check your terminal to confirm that the printed responses are accurate
4. Run test using: `npm test`

#### Any background context you want to provide?
Previously, this functionality was implemented by filtering records by updated_at date. Since new information at hand reveals that placements updates are being added as new records in allocations and not updates, hence the need for this change

#### What are the relevant pivotal tracker stories?
[#161914907](https://www.pivotaltracker.com/story/show/161914907)

#### Postman Documentation
N/A
